test/integration/component/test_template_access_across_domains.py

```diff
@@ -0,0 +1,626 @@
+                        continue
+                    raise Exception(
+                        "Error in downloading template: status - %s" %
+                        template.status)
+
+            else:
+                retries = retries - 1
+        raise Exception("Template download failed exception.")
+
+    def list_templates(self, templatefilter, domain):
+        return Template.list(
+                    self.apiclient,
+                    templatefilter=templatefilter,
+                    zoneid=self.zone.id,
+                    domainid=domain.id)
+
+    def validate_template_ownership(self, template, owner, nonowner, include_cross_domain_template):
+        """List the template belonging to domain which created it
+           Make sure that other domain can't access it.
+        """
+        list_template_response = self.list_templates('all', owner)
+        if list_template_response is not None:
+            """If global setting is false then public templates of any domain should
+               be accessible by any other domain
+            """
+            if include_cross_domain_template:
+                for temp in list_template_response:
+                    if template.name == temp.name:
+                        return
+
+                raise Exception("Template %s belonging to domain %s should "
+                                "be accessible by domain %s"
+                                % (template.name, nonowner.name, owner.name))
+            else:
+                """If global setting is true then public templates of any domain should not
+                   be accessible by any other domain except for root domain
+                """
+                for temp in list_template_response:
+                    if template.name == temp.name:
+                        raise Exception("Template %s belonging to domain %s should "
+                                        "not be accessible by domain %s"
+                                        % (template.name, nonowner.name, owner.name))
+
+    def get_restrict_template_configuration(self, domain_id):
+        """
+        Function to get the global setting "restrict.public.access.to.templates" for domain
+        """
+        list_configurations_cmd = listConfigurations.listConfigurationsCmd()
+        list_configurations_cmd.name = "restrict.public.template.access.to.domain"
+        list_configurations_cmd.scopename = "domain"
+        list_configurations_cmd.scopeid = domain_id
+        response = self.apiclient.listConfigurations(list_configurations_cmd)
+        return response[0].value
+
+    def update_restrict_template_configuration(self, domain_id, value):
+        """
+        Function to update the global setting "restrict.public.access.to.templates" for domain
+        """
+        update_configuration_cmd = updateConfiguration.updateConfigurationCmd()
+        update_configuration_cmd.name = "restrict.public.template.access.to.domain"
```

Comment:
@soreana 
since you have changed the name of global configuration to `share.public.templates.with.other.domains`, can you change the name in component test ? (maybe the value as well)